website/js/index.js

```diff
@@ -506,20 +499,31 @@ var VariantDetail = React.createClass({
                 // we always want reports grouped by source, so we'll do so centrally here
                 const groupedReports = _.groupBy(resp.data, 'Source');
 
-                return this.setState({reports: groupedReports, error: null});
+                this.setState({reports: groupedReports, error: null}, () => {
+                    this.relayoutGrid();
+                });
             }, () => {
                 this.setState({reportError: 'Problem retrieving reports'});
                 console.warn("Couldn't retrieve reports!");
             }
         );
+
     },
-    onChildToggleMode: function() {
-        this.props.toggleMode();
-        // we need to reparse the tooltips if the mode changed
-        this.setState({
-            tooltips: parseTooltips(localStorage.getItem("research-mode") === 'true')
-        });
-        this.forceUpdate();
+    componentWillUpdate: function(nextProps) {
```

Comment:
onChildToggleMode was removed because the only thing it did aside from relayout the tiles via a forced update was set the tooltips, and this call was being skipped by the new mode changer in the navigation bar anyway. Changing it to componentWillUpdate (for the state-changing parts) and componentDidUpdate (for the tile relayout, which just involves the DOM) ensures that we always catch and appropriately handle the mode change.

Reply:
nice!